django/views/i18n.py

```diff
@@ -94,87 +94,87 @@ def get_formats():
   django.pluralidx = function (count) { return (count == 1) ? 0 : 1; };
   {% endif %}
 
-  {% if catalog_str %}
   /* gettext library */
 
-  django.catalog = {{ catalog_str }};
-
-  django.gettext = function (msgid) {
-    var value = django.catalog[msgid];
-    if (typeof(value) == 'undefined') {
-      return msgid;
-    } else {
-      return (typeof(value) == 'string') ? value : value[0];
-    }
-  };
-
-  django.ngettext = function (singular, plural, count) {
-    var value = django.catalog[singular];
-    if (typeof(value) == 'undefined') {
-      return (count == 1) ? singular : plural;
-    } else {
-      return value[django.pluralidx(count)];
-    }
-  };
-
-  django.gettext_noop = function (msgid) { return msgid; };
-
-  django.pgettext = function (context, msgid) {
-    var value = django.gettext(context + '\x04' + msgid);
-    if (value.indexOf('\x04') != -1) {
-      value = msgid;
-    }
-    return value;
-  };
-
-  django.npgettext = function (context, singular, plural, count) {
-    var value = django.ngettext(context + '\x04' + singular, context + '\x04' + plural, count);
-    if (value.indexOf('\x04') != -1) {
-      value = django.ngettext(singular, plural, count);
-    }
-    return value;
-  };
-  {% else %}
-  /* gettext identity library */
-
-  django.gettext = function (msgid) { return msgid; };
-  django.ngettext = function (singular, plural, count) { return (count == 1) ? singular : plural; };
-  django.gettext_noop = function (msgid) { return msgid; };
-  django.pgettext = function (context, msgid) { return msgid; };
-  django.npgettext = function (context, singular, plural, count) { return (count == 1) ? singular : plural; };
+  django.catalog = django.catalog || {};
+  {% if catalog_str %}
+  var newcatalog = {{ catalog_str }};
+  for (var key in newcatalog) {
+    django.catalog[key] = newcatalog[key];
+  }
   {% endif %}
 
-  django.interpolate = function (fmt, obj, named) {
-    if (named) {
-      return fmt.replace(/%\(\w+\)s/g, function(match){return String(obj[match.slice(2,-2)])});
-    } else {
-      return fmt.replace(/%s/g, function(match){return String(obj.shift())});
-    }
-  };
-
-
-  /* formatting library */
-
-  django.formats = {{ formats_str }};
-
-  django.get_format = function (format_type) {
-    var value = django.formats[format_type];
-    if (typeof(value) == 'undefined') {
-      return format_type;
-    } else {
+  if (!django.jsi18n_initialized) {
+    django.gettext = function (msgid) {
```

Comment:
I think we should drop the space after "function" through this file (can be a separate commit).